langfuse-core/src/utils.ts

```diff
@@ -90,3 +92,48 @@ export function getEnv<T = string>(key: string): T | undefined {
   }
   return;
 }
+
+interface Params extends LangfuseCoreOptions {
+  publicKey?: string;
+  secretKey?: string;
+}
+
+export function configLangfuseSDK(params?: Params, secretRequired: boolean = true): Params {
+  if (!params) {
+    params = {};
+  }
+  console.log("params", params, secretRequired);
+  const { publicKey, secretKey, ...coreOptions } = params;
+
+  // check environment variables if values not provided
+  const finalPublicKey = publicKey ?? getEnv("LANGFUSE_PUBLIC_KEY");
+  const finalSecretKey = secretRequired ? secretKey ?? getEnv("LANGFUSE_SECRET_KEY") : undefined;
+  const finalBaseUrl = coreOptions.baseUrl ?? getEnv("LANGFUSE_BASEURL");
+
+  console.log("finalBaseUrl", coreOptions.baseUrl, getEnv("LANGFUSE_BASEURL"));
+  const finalCoreOptions = {
+    ...coreOptions,
+    baseUrl: finalBaseUrl,
+  };
+
+  console.log("finalCoreOptions", finalCoreOptions, {
+    publicKey: finalPublicKey,
+    ...(secretRequired === true ? { secretKey: finalSecretKey } : undefined),
+    ...finalCoreOptions,
+  });
+
+  // check required parameters
+  if (!finalPublicKey) {
+    console.error("publicKey is required, but was not provided");
```

Comment:
Improved error message, hint at that it can be set via constructor argument or environment secret